in_toto/models/metadata.py

```diff
@@ -395,18 +399,28 @@ def verify_signature(self, verification_key):
                 "No signature found for key '{}'".format(verification_keyid)
             )
 
+        valid = False
         if securesystemslib.formats.GPG_SIGNATURE_SCHEMA.matches(signature):
             valid = securesystemslib.gpg.functions.verify_signature(
                 signature, verification_key, self.signed.signable_bytes
             )
 
-        elif securesystemslib.formats.SIGNATURE_SCHEMA.matches(signature):
-            valid = securesystemslib.keys.verify_signature(
-                verification_key, signature, self.signed.signable_bytes
-            )
-
         else:
-            valid = False
+            # Parse key and (below) signature dicts as `Key` and `Signature`
+            # instances to use modern securesystemslib verification code.
+            # Deepcopy to preserve original dicts, which are otherwise destroyed
```

Comment:
Hmm, interesting. Do the `from_dict` methods _need_ to be destructive?

Reply:
This aligns with usage in python-tuf's Metadata API, where dicts can be discarded once deserialised.

There it turned out useful to just pop all expected fields and cram whatever's left into  a `unrecognized_fields: Optional[Dict[str, Any]]` attribute, which exists on all classes that have a `from_dict` deserialisation helper.

Unfortunately, this doesn't work in in-toto, where we only have classes for parts of the metadata but keep most contained data (signatures, keys, etc.) as dicts.

That's what I meant with supporting Key and Signature natively being nice. I don't think it's a priority, but I am happy to submit an issue.

Reply:
Oh, I think I see, hmm. Still don't understand why we can't hide away destruction (e.g., by copying) on behalf of users, but you're right, this is not a high-priority issue, although we should make an issue just to be safe. Thanks!